### DISCUSSION TITLE
How do I make a LineChart with x/y data? How to scale the axes?

### DISCUSSION THREAD

**Discussion by CaiusJard** [Megabit/Blazorise#2224] [Q&A]
So, I'm looking at chartjs's docs and trying to match them up against this lib's docs because I can't find the info I need in the Blazorise Charts doc...

It seems that chartjs would hopefully allow me to put data like:

    data: [{x:'2021-01-01T12:00:00Z', y:20}, {x:'2021-01-01T12:10:00Z', y:10}, {x:'2021-01-01T12:50:00Z', y:30}]

And somehow declare the axis as date, so those values get parsed correctly, and then I could have a line that steps up and down between these intervals (which do not occur regularly)

The Blazorise Chart docs seem to use Primitive type dataset e.g. `LineChartDataSet<double>` but I'm wondering how/if I can use the x/y flavor instead, and if so, what does that look like? Will any object that has an X and Y property (like `(DateTime x, double y)` or anonymous type work, or must it be a dedicated class? and is there a dedicated class already in blazorise?  

I also assume that there's a way to declare the type of the x axis, or will it be detected from the type I pass for X in the dataset?

**Comment by stsrki** [comment]
Can this help you? https://github.com/stsrki/Blazorise/discussions/2127

**Reply by CaiusJard** [reply]
It does some, yes.. It looks like the answer to my question about data is "use a type that has X and Y".. But can it work with an axis that is DateTime type? Or must I transform my DateTimes to a float/double?

**Reply by stsrki** [reply]
It can work with any data. You will need to create some custom `struct` with the DateTime field. For example in our live demo you can see the similar thing https://github.com/stsrki/Blazorise/blob/master/Demos/Blazorise.Demo/Pages/Tests/LiveChartsPage.razor.cs#L353

**Reply by CaiusJard** [reply]
Ok, I'll give a try! :) Thanks for being so quick to respond, excellent help!

**Reply by stsrki** [reply]
No problem. I try to answer whenever I can. Sometimess it can take a little longer 😅
